### ISSUE TITLE
[ML] Failures in mixed cluster BWC tests for data frame transforms crud

### ISSUE THREAD

**Issue by dimitris-athanasiou** [elastic/elasticsearch#44808]
Build link: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=zulu11,nodes=general-purpose/100/console

Reproduce:

```
./gradlew :x-pack:qa:rolling-upgrade:v7.3.0#twoThirdsUpgradedTestRunner --tests "org.elasticsearch.upgrades.UpgradeClusterClientYamlTestSuiteIT.test {p0=mixed_cluster/80_data_frame_jobs_crud/Test GET, start, and stop old cluster batch transforms}" -Dtests.seed=CEB38F680E8C42A6 -Dtests.security.manager=true -Dtests.locale=fr-BF -Dtests.timezone=Asia/Qyzylorda -Dcompiler.java=12 -Druntime.java=11 -Dtests.rest.suite=mixed_cluster
```

Failure:

```
org.elasticsearch.upgrades.UpgradeClusterClientYamlTestSuiteIT > test {p0=mixed_cluster/80_data_frame_jobs_crud/Test GET, start, and stop old cluster batch transforms} FAILED
    java.lang.AssertionError: Failure at [mixed_cluster/80_data_frame_jobs_crud:189]: expected [2xx] status code but api [data_frame.stop_data_frame_transform] returned [409 Conflict] [{"error":{"root_cause":[{"type":"status_exception","reason":"Unable to stop data frame transform [old-complex-transform] as it is in a failed state with reason [Failed to load data frame transform state for transform [old-complex-transform]]. Use force stop to stop the data frame transform.","stack_trace":"ElasticsearchStatusException[Unable to stop data frame transform [old-complex-transform] as it is in a failed state with reason [Failed to load data frame transform state for transform [old-complex-transform]]. Use force stop to stop the data frame transform.]\n\tat org.elasticsearch.xpack.dataframe.action.TransportStopDataFrameTransformAction.validateTaskState(TransportStopDataFrameTransformAction.java:93)\n\tat org.elasticsearch.xpack.dataframe.action.TransportStopDataFrameTransformAction.lambda$doExecute$0(TransportStopDataFrameTransformAction.java:123)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)\n\tat org.elasticsearch.xpack.dataframe.persistence.DataFrameTransformsConfigManager.lambda$expandTransformIds$6(DataFrameTransformsConfigManager.java:244)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)\n\tat org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:43)\n\tat org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:68)\n\tat org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:64)\n\tat org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:43)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.sendSearchResponse(AbstractSearchAsyncAction.java:300)\n\tat org.elasticsearch.action.search.FetchSearchPhase$3.run(FetchSearchPhase.java:213)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:171)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:165)\n\tat org.elasticsearch.action.search.ExpandSearchPhase.run(ExpandSearchPhase.java:119)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:171)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:165)\n\tat org.elasticsearch.action.search.FetchSearchPhase.moveToNextPhase(FetchSearchPhase.java:206)\n\tat org.elasticsearch.action.search.FetchSearchPhase.lambda$innerRun$2(FetchSearchPhase.java:104)\n\tat org.elasticsearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:110)\n\tat org.elasticsearch.action.search.FetchSearchPhase.access$000(FetchSearchPhase.java:44)\n\tat org.elasticsearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:86)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:773)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.lang.Thread.run(Thread.java:834)\n"}],"type":"status_exception","reason":"Unable to stop data frame transform [old-complex-transform] as it is in a failed state with reason [Failed to load data frame transform state for transform [old-complex-transform]]. Use force stop to stop the data frame transform.","stack_trace":"ElasticsearchStatusException[Unable to stop data frame transform [old-complex-transform] as it is in a failed state with reason [Failed to load data frame transform state for transform [old-complex-transform]]. Use force stop to stop the data frame transform.]\n\tat org.elasticsearch.xpack.dataframe.action.TransportStopDataFrameTransformAction.validateTaskState(TransportStopDataFrameTransformAction.java:93)\n\tat org.elasticsearch.xpack.dataframe.action.TransportStopDataFrameTransformAction.lambda$doExecute$0(TransportStopDataFrameTransformAction.java:123)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)\n\tat org.elasticsearch.xpack.dataframe.persistence.DataFrameTransformsConfigManager.lambda$expandTransformIds$6(DataFrameTransformsConfigManager.java:244)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)\n\tat org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:43)\n\tat org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:68)\n\tat org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:64)\n\tat org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:43)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.sendSearchResponse(AbstractSearchAsyncAction.java:300)\n\tat org.elasticsearch.action.search.FetchSearchPhase$3.run(FetchSearchPhase.java:213)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:171)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:165)\n\tat org.elasticsearch.action.search.ExpandSearchPhase.run(ExpandSearchPhase.java:119)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:171)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:165)\n\tat org.elasticsearch.action.search.FetchSearchPhase.moveToNextPhase(FetchSearchPhase.java:206)\n\tat org.elasticsearch.action.search.FetchSearchPhase.lambda$innerRun$2(FetchSearchPhase.java:104)\n\tat org.elasticsearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:110)\n\tat org.elasticsearch.action.search.FetchSearchPhase.access$000(FetchSearchPhase.java:44)\n\tat org.elasticsearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:86)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:773)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.lang.Thread.run(Thread.java:834)\n"},"status":409}]

        Caused by:
        java.lang.AssertionError: expected [2xx] status code but api [data_frame.stop_data_frame_transform] returned [409 Conflict] [{"error":{"root_cause":[{"type":"status_exception","reason":"Unable to stop data frame transform [old-complex-transform] as it is in a failed state with reason [Failed to load data frame transform state for transform [old-complex-transform]]. Use force stop to stop the data frame transform.","stack_trace":"ElasticsearchStatusException[Unable to stop data frame transform [old-complex-transform] as it is in a failed state with reason [Failed to load data frame transform state for transform [old-complex-transform]]. Use force stop to stop the data frame transform.]\n\tat org.elasticsearch.xpack.dataframe.action.TransportStopDataFrameTransformAction.validateTaskState(TransportStopDataFrameTransformAction.java:93)\n\tat org.elasticsearch.xpack.dataframe.action.TransportStopDataFrameTransformAction.lambda$doExecute$0(TransportStopDataFrameTransformAction.java:123)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)\n\tat org.elasticsearch.xpack.dataframe.persistence.DataFrameTransformsConfigManager.lambda$expandTransformIds$6(DataFrameTransformsConfigManager.java:244)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)\n\tat org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:43)\n\tat org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:68)\n\tat org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:64)\n\tat org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:43)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.sendSearchResponse(AbstractSearchAsyncAction.java:300)\n\tat org.elasticsearch.action.search.FetchSearchPhase$3.run(FetchSearchPhase.java:213)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:171)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:165)\n\tat org.elasticsearch.action.search.ExpandSearchPhase.run(ExpandSearchPhase.java:119)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:171)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:165)\n\tat org.elasticsearch.action.search.FetchSearchPhase.moveToNextPhase(FetchSearchPhase.java:206)\n\tat org.elasticsearch.action.search.FetchSearchPhase.lambda$innerRun$2(FetchSearchPhase.java:104)\n\tat org.elasticsearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:110)\n\tat org.elasticsearch.action.search.FetchSearchPhase.access$000(FetchSearchPhase.java:44)\n\tat org.elasticsearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:86)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:773)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.lang.Thread.run(Thread.java:834)\n"}],"type":"status_exception","reason":"Unable to stop data frame transform [old-complex-transform] as it is in a failed state with reason [Failed to load data frame transform state for transform [old-complex-transform]]. Use force stop to stop the data frame transform.","stack_trace":"ElasticsearchStatusException[Unable to stop data frame transform [old-complex-transform] as it is in a failed state with reason [Failed to load data frame transform state for transform [old-complex-transform]]. Use force stop to stop the data frame transform.]\n\tat org.elasticsearch.xpack.dataframe.action.TransportStopDataFrameTransformAction.validateTaskState(TransportStopDataFrameTransformAction.java:93)\n\tat org.elasticsearch.xpack.dataframe.action.TransportStopDataFrameTransformAction.lambda$doExecute$0(TransportStopDataFrameTransformAction.java:123)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)\n\tat org.elasticsearch.xpack.dataframe.persistence.DataFrameTransformsConfigManager.lambda$expandTransformIds$6(DataFrameTransformsConfigManager.java:244)\n\tat org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:62)\n\tat org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:43)\n\tat org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:68)\n\tat org.elasticsearch.action.support.TransportAction$1.onResponse(TransportAction.java:64)\n\tat org.elasticsearch.action.support.ContextPreservingActionListener.onResponse(ContextPreservingActionListener.java:43)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.sendSearchResponse(AbstractSearchAsyncAction.java:300)\n\tat org.elasticsearch.action.search.FetchSearchPhase$3.run(FetchSearchPhase.java:213)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:171)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:165)\n\tat org.elasticsearch.action.search.ExpandSearchPhase.run(ExpandSearchPhase.java:119)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:171)\n\tat org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:165)\n\tat org.elasticsearch.action.search.FetchSearchPhase.moveToNextPhase(FetchSearchPhase.java:206)\n\tat org.elasticsearch.action.search.FetchSearchPhase.lambda$innerRun$2(FetchSearchPhase.java:104)\n\tat org.elasticsearch.action.search.FetchSearchPhase.innerRun(FetchSearchPhase.java:110)\n\tat org.elasticsearch.action.search.FetchSearchPhase.access$000(FetchSearchPhase.java:44)\n\tat org.elasticsearch.action.search.FetchSearchPhase$1.doRun(FetchSearchPhase.java:86)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:44)\n\tat org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:773)\n\tat org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37)\n\tat java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.lang.Thread.run(Thread.java:834)\n"},"status":409}]
```

**Comment by droberts195** [comment]
It doesn't make sense to only run a single test in the BWC tests because no cleanup takes place between tests and the later tests assume the earlier ones have run first.  Therefore a better repro command is:

```
./gradlew :x-pack:qa:rolling-upgrade:v7.3.0#bwcTest -Dtests.seed=CEB38F680E8C42A6 -Dtests.locale=fr-BF -Dtests.timezone=Asia/Qyzylorda -Dcompiler.java=12 -Druntime.java=11
```

I ran this on the 7.4.0 branch after unmuting the test locally and it didn't fail.

**Comment by hendrikmuhs** [comment]
fixed with #46595 and #46452
